controller/replicaset_test.go

```diff
@@ -249,23 +240,23 @@ func TestReconcileOldReplicaSet(t *testing.T) {
 			oldRS.Annotations = map[string]string{annotations.DesiredReplicasAnnotation: strconv.Itoa(test.oldReplicas)}
 			oldRS.Status.AvailableReplicas = int32(test.readyPodsFromOldRS)
 			oldRSs := []*appsv1.ReplicaSet{oldRS}
-			allRSs := []*appsv1.ReplicaSet{oldRS, newRS}
 			rollout := newBlueGreenRollout("foo", test.rolloutReplicas, nil, "", "")
 			rollout.Spec.Selector = &metav1.LabelSelector{MatchLabels: newSelector}
-			fake := fake.Clientset{}
-			k8sfake := k8sfake.Clientset{}
-			controller := &Controller{
-				rolloutsclientset: &fake,
-				kubeclientset:     &k8sfake,
-				recorder:          &record.FakeRecorder{},
-			}
-			scaled, err := controller.reconcileOldReplicaSets(allRSs, oldRSs, newRS, rollout)
+
```

Comment:
Prior to changes, test was returning an empty replicaset instead of a modified rollout when the Update Rollout command was called. This caused a panic when the replicas pointer was dereferenced with the new changes. We needed to start the informers in order to have the reactor react on the update call and return the correct updated object.